soc/arm/nxp_imx/Kconfig

```diff
@@ -23,4 +23,15 @@ config SOC_PART_NUMBER
 	default SOC_PART_NUMBER_IMX8ML_M7 if SOC_SERIES_IMX8ML_M7
 	default SOC_PART_NUMBER_IMX8MQ_M4 if SOC_SERIES_IMX8MQ_M4
 
+config FLASH_MCUX_FLEXSPI_XIP
+	bool "MCUX FlexSPI flash access with xip"
+	default $(dt_node_has_compat, $(DT_CHOSEN_FLASH_PARENT), $(DT_COMPAT_FLEXSPI))
```

Comment:
We still need this KConfig to be enabled for the RT1xxx series. I have not verified that the `zephyr,flash` chosen node is the 
SPI flash chip for all RT boards. If this KConfig isn't set, flash samples will likely fail, as the flash driver will attempt to reconfigure the FlexSPI while running in XIP, which will likely crash the application. Have you tested this change with a flash sample on a R6xx, RT5xx, and RT1xxx board?

Reply:
I have not tested these SOC's yet, I can test real quick with the RT595